Reef: filestore removal

NitzanMordhai opened the pull request:
FileStore depreciation for Reef - Updated or new installed OSDs will no longer be able to start with FileStore.
Since FileStore will be depreciated, we have few tests for FileStore that can be removed and some tools that need to be adjusting to use BlueStore or remove some functionality that no longer needed.





<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

aclamk commented on qa/suites/rados/objectstore/backends/alloc-hint.yaml:
```diff
@@ -11,7 +11,7 @@ overrides:
         filestore xfs extsize: true
-        osd objectstore: filestore
+        osd objectstore: bluestore
         osd op queue: wpq
```
I think this file should just be removed.

NitzanMordhai replied:
That means I can remove also test_alloc_hint.sh ?

aclamk commented on qa/suites/rados/objectstore/backends/ceph_objectstore_tool.yaml:
```diff
@@ -13,7 +13,7 @@ tasks:
       osd:
-        osd objectstore: filestore
+        osd objectstore: bluestore
         osd op queue: wpq
```
I think entire file should be removed. 
And qa/suites/rados/objectstore/backends/filejournal.yaml too.

NitzanMordhai replied:
removing filejournal.yaml
ceph_objectstore_tool.yaml - but there are still functions that left there that are not related to filestore, should we still consider removing it?

NitzanMordhai replied:
@aclamk ?

aclamk replied:
You are correct, lets leave it as is.

NitzanMordhai replied:
@aclamk Thanks!

aclamk commented on src/test/objectstore/store_test.cc:
```diff
@@ -6839,7 +6838,6 @@ INSTANTIATE_TEST_SUITE_P(
   StoreTest,
   ::testing::Values(
     "memstore",
-    "filestore",
```
This change requires shifting of --gtest_filter parametes in 
qa/suites/rados/objectstore/backends/* files where ceph_test_objectstore is called.
For example `--gtest_filter=*/2:-*SyntheticMatrixC*` no longer means bluestore but kstore.

NitzanMordhai replied:
Done

rzarzynski commented on qa/tasks/ceph.py:
```diff
@@ -1585,6 +1585,14 @@ def restart(ctx, config):
+		    args = [
+                        'sudo',
+                        'egrep', pattern,
+                        '/var/log/ceph/{cluster}.log'.format(cluster=cluster),
```
I doubt we'll have an access the monitor at time when OSD calls `ObjectStore::create()` – `MonClient` is instantiated later.

NitzanMordhai replied:
Changing to egrep from osds logs

rzarzynski commented:
22+k SLOCs removed :-D.

NitzanMordhai commented:
i modified the ceph.py a bit to check during restart if there is any expected failure, run some cases and removed dead test for filestore: http://pulpito.front.sepia.ceph.com/nmordech-2022-12-28_11:48:43-upgrade:quincy-x:filestore-remove-check-wip-nitzan-filestore-removal-distro-default-smithi/

NitzanMordhai commented:
jenkins test make check

NitzanMordhai commented:
jenkins test make check

NitzanMordhai commented:
jenkins test make check

wjwithagen commented:
This means that from this moment on releases will not be able to run on FreeBSD.
Patches to get Bluestore in for FreeBSD have been lingering for ever...
I've been quiet for some time already, since my real job has moved on.
I do still lurk, but have little to no time to work on it.

That said, it would be sad if that code would rot. Fortunatly some other ports are using some parts of the infrastructure.
So I guess not everything was futile.

rzarzynski commented on qa/suites/rados/objectstore/backends/objectstore-bluestore-a.yaml:
```diff
@@ -8,5 +8,5 @@ tasks:
 - exec:
     client.0:
-      - mkdir $TESTDIR/archive/ostest && cd $TESTDIR/archive/ostest && ulimit -Sn 16384 && CEPH_ARGS="--no-log-to-stderr --log-file $TESTDIR/archive/ceph_test_objectstore.log --debug-bluestore 20" ceph_test_objectstore --gtest_filter=*/2:-*SyntheticMatrixC* --gtest_catch_exceptions=0
+      - mkdir $TESTDIR/archive/ostest && cd $TESTDIR/archive/ostest && ulimit -Sn 16384 && CEPH_ARGS="--no-log-to-stderr --log-file $TESTDIR/archive/ceph_test_objectstore.log --debug-bluestore 20" ceph_test_objectstore --gtest_filter=*/1:-*SyntheticMatrixC* --gtest_catch_exceptions=0
```
ACK.

NitzanMordhai commented:
jenkins test make check

idryomov reviewed (commented):
Some leftovers that I have spotted:
- `hadoop` suite has a dangling `filestore-xfs.yaml` symlink (not sure if anyone ran this suite in recent years though)
- `ceph_volume` suite has a big number of `filestore` facets which pass `osd_objectstore: "filestore"`
- https://docs.ceph.com/en/latest/rados/configuration/filestore-config-ref/ page needs to go ("... though Filestore OSDs are still supported" claim)

NitzanMordhai commented:
> * `hadoop` suite has a dangling `filestore-xfs.yaml` symlink (not sure if anyone ran this suite in recent years though)
  
Removed
> 
> * `ceph_volume` suite has a big number of `filestore` facets which pass `osd_objectstore: "filestore"`
I didn't even notice there are suites over that path! Thanks Removed!

NitzanMordhai commented:
> * https://docs.ceph.com/en/latest/rados/configuration/filestore-config-ref/ page needs to go ("... though Filestore OSDs are still supported" claim)

It looks like the docs will need extra care, how about letting that PR pass and then handle the docs? i'll open new PR for it, if thats ok with you all @idryomov @rzarzynski @idryomov @aclamk @neha-ojha

idryomov commented:
> It looks like the docs will need extra care, how about letting that PR pass and then handle the docs?

Sure, a separate docs PR to remove all references to Filestore sounds good to me.

BTW there are now over 20 commits in this PR.  Please squash all pure code removal commits, if a few lines are adjusted in some test case it doesn't need a separate commit either, etc.

NitzanMordhai commented:
> BTW there are now over 20 commits in this PR. Please squash all pure code removal commits, if a few lines are adjusted in some test case it doesn't need a separate commit either, etc.

Done, thanks!

idryomov commented:
Was "Rados\suites: change all osd objectstore filestore" meant to contain a backslash?  I'd rename it to "qa/suites: change all osd objectstore filestore" and merge the ceph-volume commit into it as well (since it's also a suite).

NitzanMordhai commented:
> Was "Rados\suites: change all osd objectstore filestore" meant to contain a backslash? I'd rename it to "qa/suites: change all osd objectstore filestore" and squash the ceph-volume commit into it as well (since it's also a suite).

Done

rzarzynski commented on src/os/ObjectStore.cc:
```diff
@@ -54,8 +54,9 @@ std::unique_ptr<ObjectStore> ObjectStore::create(
-  if (type == "filestore" || (type == "random" && rand() % 2)) {
-    return std::make_unique<FileStore>(cct, data, journal, flags);
+  if (type == "filestore") {
+    lgeneric_derr(cct) << __func__ << ": FileStore has been deprecated and is no longer supported" << dendl;
```
Bump-up.

NitzanMordhai commented:
jenkins test api

NitzanMordhai commented:
jenkins test api

idryomov reviewed (approved):
Ack from the RBD perspective!

We do have some Filestore-specific bits in librbd such as below but they are staying for a long time to come for handling older OSDs.
https://github.com/ceph/ceph/blob/01ad87ef30f99cae14baa152cc0b7bdf6ec0a114/src/librbd/io/ImageRequest.cc#L619-L634
